PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/SparkScalaLivyConsoleRunProfileState.kt

```diff
@@ -49,25 +54,25 @@ class SparkScalaLivyConsoleRunProfileState(private val consoleBuilder: SparkScal
 
     override fun execute(executor: Executor, runner: ProgramRunner<*>): ExecutionResult? {
         val console = consoleBuilder.console
-
-        val livySessionProcessHandler = SparkLivySessionProcessHandler(SparkLivySessionProcess(session))
+        val progressBarScheduler = IdeaSchedulers(consoleBuilder.project)
+        val livySessionProcess = SparkLivySessionProcess(progressBarScheduler, session)
+        val livySessionProcessHandler = SparkLivySessionProcessHandler(livySessionProcess)
 
         console.attachToProcess(livySessionProcessHandler)
+        logSubject.subscribe({ typedMessage -> console.print(typedMessage) }, { err ->
+            console.print("Livy interactive session is stop due to the error ${err.message}", LOG_ERROR_OUTPUT)
+        })
 
-        val progressBarScheduler = IdeaSchedulers(consoleBuilder.project)
-                .processBarVisibleAsync("Start Spark Livy interactive console session...")
-
-        session.create()
-                .subscribeOn(progressBarScheduler)
-                .flatMap { it.awaitReady(progressBarScheduler) }
+        livySessionProcess
+                .start()
                 .subscribe({
                     livySessionProcessHandler.execute(postStartCodes)
                     (console as? ScalaLanguageConsole)?.apply {
                         // Customize the Spark Livy interactive console
                         prompt = "\nSpark>"
                     }
                 }, { ex ->
-                    console.print("Can't start Livy interactive session: ${ex.message}\n", ConsoleViewContentType.LOG_ERROR_OUTPUT)
+                    console.print("Can't start Livy interactive session: ${ex.message}\n", LOG_ERROR_OUTPUT)
```

Comment:
Maybe we can replace `console.print` with `logSubject.onNext`

Reply:
Good suggestion!